File: 349. Intersection of Two Arrays/intersection.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
const intersection = (nums1, nums2) => {
  const res = [];
  const nums1Set = new Set(nums1);

  for (let i = 0; i < nums2.length; i++) {
    if (nums1Set.has(nums2[i])) {
      res.push(nums2[i]);
      nums1Set.delete(nums2[i]);
    }
  }

  return res;
};

const nums1 = [1, 2, 2, 1];
const nums2 = [2, 2];
console.log('intersection: ', intersection(nums1, nums2));

File: 349. Intersection of Two Arrays/intersection.py
from typing import List

class Solution:
  def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
    res = []
    nums1Set = set(nums1)

    for num in nums2:
      if num in nums1Set:
        res.append(num)
        nums1Set.remove(num)

    return res


nums1 = [1, 2, 2, 1]
nums2 = [2, 2]
solution = Solution()
print('intersection: ', solution.intersection(nums1, nums2))
